app/build_svc.py

```diff
@@ -19,78 +20,180 @@ def __init__(self, services):
-                                                      command=('%s %s' % (env['entrypoint'], args)).split(' '),
+                                                      command='{} {}'.format(env['entrypoint'], args).split(' '),
                                                       working_dir=env['workdir'],
+                                                      user=os.getuid(),
                                                       volumes={os.path.abspath(
                                                                os.path.join(self.build_directory, ability.language)):
                                                                dict(bind=env['workdir'], mode='rw')}, detach=True)
         code = container.wait()
-        self.log.debug('Container for %s ran for ability ID %s: %s' % (ability.language, ability.ability_id,
-                                                                       code))
+        self.log.debug('Container for {} ran for ability ID {}: {}'.format(ability.language, ability.ability_id, code))
+
+    def _check_errors(self, language):
+        """Check for errors which occurred during the build
+
+        :param language: Language to check errors for
+        :type language: string
+        """
+        if language == 'csharp':
+            error_log = os.path.join(self.build_directory, language, 'error.log')
+            if os.path.isfile(error_log):
+                with open(error_log) as f:
+                    log_data = json.load(f)
+
+                errors = log_data['runs'][0]['results']
+                for error in errors:
+                    location_data = ''
+                    locations = error.get('locations', [])
+                    if locations:
+                        region = locations[0]['resultFile']['region']
+                        location_data = '{}({},{},{},{}): '.format(locations[0]['resultFile']['uri'],
+                                                                   region.get('startLine'), region.get('startColumn'),
+                                                                   region.get('endLine'), region.get('endColumn'))
+                    self.log.debug('{}{} {}: {}'.format(location_data, error.get('level').capitalize(),
+                                                        error.get('ruleId'), error.get('message')))
 
     def _replace_args_props(self, ability):
+        """Replace template arguments in build command
+
+        :param ability: Ability to create command for
+        :type ability: Ability
+        :return: Created command string
+        :rtype: string
+        """
         env = self.get_config(prop='enabled', name='build').get(ability.language)
         cmd = env['entrypoint_args'].replace('#{code}', self.build_file)
         cmd = cmd.replace('#{build_target}', ability.build_target)
+
+        references = [p for p in ability.payloads if p.endswith('.dll')]
+        reference_cmd = ' -r:{}'.format(','.join(references)) if references else ''
```

Comment:
Change .format to use % 
This is to ensure we are consistent across the CALDERA code base